src/middlewared/middlewared/plugins/system.py

```diff
@@ -161,79 +164,85 @@ class Config:
 
         `legacy_ui` is disabled by default. Enabling it allows end users to use the legacy UI.
         """
-        config_data = await self.config()
-        original_data = config_data.copy()
-        config_data.update(data)
+        config_current = await self.config()
```

Comment:
If you want to refactor this, it seems to me that we are using `old` and `new` in most of our other code